enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,296 @@
+  DeprecatedJenkinsPipeline: newDefaultFeatures().
+    with("BuildJenkinsPipelineStrategy").
+    toFeatures(),
+}
+
+```
+
+The operator for openshift-controller-manager will be enhanced to watch the cluster feature gates.
+If the cluster has the `BuildJenkinsPipelineStrategy` feature gate enabled, this gate will be passed to openshift-controller-manager's configuration.
+Openshift-controller-manager will then enable the strategy and allow JenkinsPipline builds to proceed as normal.
+
+#### Bootstrap Behavior
+
+When the `BuildJenkinsPipelineStrategy` feature gate is introduced, the openshift-controller-manager-operator will query all `BuildConfig` objects to see if any utilize the JenkinsPipeline strategy.
+If any BuildConfig uses the JenkinsPipline strategy, the DeprecatedJenkinsPipeline feature set will be turned on.
+
+The bootstrap check will only run if the default feature set is configured for the cluster.
+A status condition will indicate if this bootstrap check has completed.
+Once the bootstrap check is complete, it should not be run again.
+These protections will ensure that we do not overwrite feature sets that are enabled as a day 2 action, or overwrite pre-existing feature sets that allow updgrades.
+
+#### oc <new|start>-build
+
+`oc new-build` should print a warning message if the generated/referenced BuildConfig uses the Jenkins pipeline strategy.
+This message should indicate that the cluster must have the DeprecatedJenkinsPipeline feature set enabled.
+Similarly, `oc start-build` should print a warning message if a build was started with the Jenkins pipeline strategy indicating that the cluster must have the DeprecatedJenkinsPipeline feature set enabled.
+
+#### Web Console
+
+User will continue to be able to deploy Jenkins on OpenShift by instantiating our provided Templates.
+As long as the Jenkins Pipeline strategy can be run on a fully supported cluster (i.e upgrades are allowed), existing console integrations with Jenkins should remain in place.
+
+#### Telemetry / Operational Readiness
+
+The feature set will take advantage of existing telemetry tooling.
+Clusters that enable the DeprecatedJenkinsPipeline feature set will be automatically reported to Telemeter.
+Fleet operators/SREs will be able to identify which clusters have the feature set enabled.
+
+The bootstrap behavior will ensure that most clusters which rely on the JenkinsPipeline build strategy have the feature set enabled automatically.
+The exception to this rule will be clusters that have a separate feature set enabled prior to upgrade.
+The following query in Data Hub will reveal which clusters fall into this category:
+
+`count by (_id, name) (cluster_feature_set{name!=""} offset 180m) * on (_id) group_right() openshift:build_by_strategy:sum{strategy="jenkinspipeline"} offset 180m`
+
+As of this writing, no clusters fall into this category.
+This may change if clusters opt into tech preview features related to OpenShift builds.
+
+#### Documentation
```

Comment:
Either with this documentation section or a separate section, I think we should be getting rid of https://github.com/openshift/origin/tree/master/examples/jenkins as part of the overall theme carried here on the de-prioritization and decoupling of Jenkins.

Reply:
We will need to keep Jenkins in our product documentation, so I don't want to confuse the matter here.

IMO the Jenkins example in origin can be moved to the openshift/jenkins repo, which is in line with our efforts to move Jenkins tests out of openshift/orign. I don't think we need to get sign off on this via an EP.

Reply:
Moving from openshift/origin to openshift/jenkins is better than just removing.  I think we should go with your tweak on my idea.

That said, I'm not sure I agree with it not having to be called out in the EP.  Certain elements of the openshfit community still hold value in what is under the examples subdir of openshift/origin.

But perhaps I am in the minority there.

@bparees - you have as much historical participation in the examples subdir as I do.  WDYT?

Reply:
> That said, I'm not sure I agree with it not having to be called out in the EP. Certain elements of the openshfit community still hold value in what is under the examples subdir of openshift/origin.

do they? we should be directing them to https://github.com/openshift/library/ instead.

the history of why they are there now is that we only had one repo, and also even once we started having multiple repos, tests were still expecting to get content from the origin repo.

i don't see a good reason to keep the content in origin at this point, as long as we put it in openshift/library (and openshift/library can import it from openshift/jenkins).  You get the templates as part of your OCP install anyway from the samples operator.

Reply:
That "implementation" approach works for me as well @bparees 

Now, where are you with calling all of this out in @adambkaplan 's EP?  An implementation that should be included?  Or details he does not have to add ?

Reply:
given that we've now had the conversation and defined a plan, it seems worth preserving it in the EP itself

Reply:
We already sync Jenkins imagestreams and templates from openshift/jenkins to openshift/library.

Updating the docs section to state that tests should use one of the templates delivered via the Samples operator.

Reply:
FWIW my tech debt moving of e2e's from openshift origin to the jenkins related repos will achieve the result of no longer referencing https://github.com/openshift/origin/tree/master/examples/jenkins in our jenkins related e2e's